src/Engine/ProtoCore/ProcedureTable.cs

```diff
@@ -333,57 +349,22 @@ public IEnumerable<ProcedureNode> GetFunctionsByNameAndArgumentNumber(string nam
         }
 
         // TODO: To be deleted.
+        [Obsolete("This method will be removed in Dynamo 3.0. Please use GetFunctionBySignature with ProcedureMatchOptions instead.")]
         public int IndexOf(string name, List<ProtoCore.Type> argTypeList, bool isStaticOrConstructor = false)
         {
-            int currentProcedure = ProtoCore.DSASM.Constants.kInvalidIndex;
-            bool functionPointerCheck = argTypeList == null;
-
-            if (functionPointerCheck) // check for function pointer
```

Comment:
Is this condition taken care of in the new `GetFunctionBySignature` method?

Reply:
Yes, the new opts.ParameterTypes can be null.